File: Samples/Mapping/Maps-Samples/Categories/PWBuilding+Helper.m
//
//  PWMapView+Helper.m
//  Maps-Samples
//
//  Created on 9/16/16.
//  Copyright © 2016 Phunware. All rights reserved.
//

#import "PWBuilding+Helper.h"

@implementation PWBuilding(Helper)

- (NSArray *)getAllPOIs {
    NSMutableArray *allPOIs = [NSMutableArray new];
    for (PWFloor *floor in self.floors) {
        [allPOIs addObjectsFromArray:floor.pointsOfInterest];
    }
    return allPOIs.copy;
}

- (NSArray *)getAvailablePOITypes {
    NSMutableArray *availablePOITypes = [NSMutableArray new];
    
    for (PWPointOfInterest *poi in [self getAllPOIs]) {
        if (![availablePOITypes containsObject:poi.pointOfInterestType]) {
            [availablePOITypes addObject:poi.pointOfInterestType];
        }
    }
    
    NSSortDescriptor *sortDescriptor = [[NSSortDescriptor alloc] initWithKey:@"name" ascending:YES];
    [availablePOITypes sortUsingDescriptors:@[sortDescriptor]];
    
    return [availablePOITypes copy];
}

- (PWFloor *)floorForFloorID:(NSInteger)floorID {
    return [[self.floors filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"floorID == %@", @(floorID)]] firstObject];
}

@end

File: Samples/ImmediateRoute/ImmediateRoute/Extensions/PWBuilding+Helper.swift
//
//  PWBuilding+Helper.swift
//  Mapping-Sample
//
//  Created on 7/19/17.
//  Copyright © 2017 Phunware. All rights reserved.
//

import PWMapKit

extension PWBuilding {
    
    func allPOIs() -> [PWPointOfInterest] {
        var allPOIs = [PWPointOfInterest]()
        for floor in floors {
            if let floor = floor as? PWFloor, let pointsOfInterest = floor.pointsOfInterest as? [PWPointOfInterest] {
                allPOIs.append(contentsOf: pointsOfInterest)
            }
        }
        return allPOIs
    }
    
    func availablePOITypes() -> [PWPointOfInterestType] {
        var allPOITypes = Set<PWPointOfInterestType>()
        for poi in allPOIs() {
            allPOITypes.insert(poi.pointOfInterestType)
        }
        
        let sortedPOITypes = allPOITypes.sorted(by: {
            return $0.name < $1.name
        })
        return sortedPOITypes
    }
}
